src/CoreBundle/private/library/classes/TCMSURLHistory.class.php

```diff
@@ -220,7 +237,8 @@ public function paramsParameterExists()
         }
 
         if (false === isset($this->aHistory[0])) {
-            $this->aHistory = array_values($this->aHistory); // rearrange elements
+            // reset the keys to prevent key gaps
+            $this->aHistory = array_values($this->aHistory);
```

Comment:
Not sure why we ever needed this (added it myself a few years ago because the indices were wrong, but how can this happen?).

Reply:
Code says "session may contain old history element". I cannot imagine how the index gets these gaps, but i really had situations where this happened, but during the dev process. I tested the current version and could not find any situation where this happened again. So maybe it is fixed.

Reply:
Ah right. OK let's be bold and only re-add the fix if we find out it's still needed.